File: Scala/1. Two Sum.scala
object Solution {
  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    var resultArr = Array[Int]()
    var map: Map[Int, Int] = Map.empty[Int, Int]
    for(i <- 0 until nums.size){
      val rem = target - nums(i)
      if(map.contains(rem)){
        resultArr +:= i
        resultArr +:= map(rem)
      }
      else {
        map += (nums(i) -> i)
      }
    }
    resultArr
  }
}

File: Python/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for idx, value in enumerate(nums):
            if target - value in dict.keys():
                return [dict[target - value], idx]
            else:
                dict[value] = idx
